docs/adr/0004-supply-uncompiled-javascript.md

```diff
@@ -0,0 +1,65 @@
+# Supply un-compiled Javascript in NPM package
+
+* Status: proposed
+* Date: 2021-07-05
+
+Technical Story: https://github.com/hmrc/hmrc-frontend/issues/96
+
+## Context and Problem Statement
+
+We have received requests from external consumers for providing the Javascript modules
+for our frontend components in an un-compiled form. 
+
+Currently, the js:compile-hmrc Gulp task rolls up not just `all.js` but every
+single Javascript module, even though it's not intended that these individual modules are
+used independently in SCRIPT tags.
+
+Providing the modules in their raw form would provide greater flexibility
+for consumers to compile, bundle, tree-shake and minify as they see fit, rather than being forced
+to use the all.js bundle.
+
+As such, should we provide all but all.js in an un-compiled form?
+
+## Decision Drivers
+
+* We have only ever documented the inclusion of the `all.js` bundle
+  in frontend microservices.
+* It is not possible to include the individually rolled up modules in SCRIPT tags because each
+  module exports the same `HMRCFrontend` window object, that would overwrite each other.
+* The GOV.UK prototype kit's extension framework only generates SCRIPT tags for `all.js` and `init-all.js` as per
+  the [configuration](src/govuk-prototype-kit.config.json). Therefore, this change would not affect
+  prototypes.
+* Providing un-compiled Javascript for components is consistent with how we supply the 
+  corresponding Sass modules.
+
+## Considered Options
+
+* Option 1: Do nothing
+* Option 2: Stop compiling all but the `all.js` Javascript modules, provide un-compiled instead.
```

Comment:
I like the idea of additionally providing the un-compiled version, or additional targets like UMD / ES module / whatever the spread usually is for libraries, but I still think it's useful to provide the fully compiled version

Reply:
I misread this sorry, missed the but in the option 2 above!